### ISSUE TITLE
[SCREENREADER]: Community Care service type combobox is not announcing it has a popup

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#531]
## Issue
The Facility Locator third input is a typeahead for service type. This is being announced as an "edit, type of text" input to JAWS, with no indication it has a popup or typeahead suggest. This was flagged as a SC 4.1.2 violation. Screenshot attached.

## Audit Finding
* Note 9, Defect 6

## Acceptance Criteria
* As a screen reader user, I want to hear an indication this field is a combobox or has a popup when it receives keyboard or virtual cursor focus

## Environment
* Windows 10
* IE11 + JAWS
* https://staging.va.gov/find-locations/

## Possible Fixes (optional)
I am reaching out to the auditors for clarity, but I believe the fix is as simple as adding an `aria-haspopup="listbox"` attribute to the input. The current implementation is built on the ARIA 1.0 specification for comboboxes, but the auditors recommended we review the 1.1 version of the spec. I have previously tested comboboxes built to 1.1, and didn't feel it was the best value for our users.

For now, I will recommend we add the simpler fix, and revise the ticket as needed.

## WCAG or Vendor Guidance (optional)
* [Name, Role, Value: Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)
* [ARIA 1.1 Combobox with Listbox Popup Examples](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html)
* [Legacy ARIA 1.0 Combobox With List Autocomplete Example](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.0pattern/combobox-autocomplete-list.html)
* [Differences between ARIA 1.0 and 1.1: Changes to role=”combobox”](https://www.levelaccess.com/differences-aria-1-0-1-1-changes-rolecombobox/)

## Screenshots or Trace Logs
![Screen Shot 2019-07-10 at 7.46.25 AM.png](https://images.zenhubusercontent.com/5ac217b74b5806bc2bcd3fc8/db9a6ffd-2208-4594-b0f8-2b03f7b57c75)

**Comment by 1Copenut** [comment]
@alexpappasoddball I'm going to close this one as not-an-issue based on advice from our VA 508 office contact.

**Trevor: (Wednesday, July 10, 2019)**
When the user changes the setting of the Search For list box to Community Care (Non-VA Health), the Service Type field is created. This element functions as a combo box but is read as an edit field. The user is not advised of this functionality when the field receives focus. For more, see http://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html.

In this case, I reviewed the spec for ARIA 1.1 and 1.0 versions of the combobox pattern. I also read through an article by Level Access, outlining the differences in the spec: https://www.levelaccess.com/differences-aria-1-0-1-1-changes-rolecombobox/

In looking at the ARIA 1.0 version, we could add an aria-haspopup="listbox" attribute to the Search For box. This causes JAWS to read it as "Edit, type of text has popup" in IE11. I've had mixed results using the 1.1 combobox pattern with JAWS in the past, so I lean toward the current implementation with better aria markup for clarity.

**VA 508 Office: (Jul 17, 2019, 12:28 PM)**
This is not a defect. The user gets to the box and JAWs reads "type in text" when you get to it then when you type in text it says "three results are available use arrow keys to select". It is very clear for the non-sighted user. Now the error reporting on this needs some work but that is a different defect under error reporting.
